rust/xaynet-server/src/metrics/recorders/influxdb/recorder.rs

```diff
@@ -0,0 +1,66 @@
+use super::{Dispatcher, Event, InfluxDbService, Measurement, Metric, Request, Tags};
+use crate::settings::InfluxSettings;
+
+use futures::future::poll_fn;
+use tower::Service;
+use tracing::{error, warn};
+
+/// An InfluxDB metrics / events recorder.
+pub struct Recorder {
+    /// A services that dispatches the recorded metrics / events to an InfluxDB instance.
+    service: InfluxDbService,
+}
+
+impl Recorder {
+    /// Creates a new InfluxDB recorder.
+    pub fn new(settings: InfluxSettings) -> Self {
+        let dispatcher = Dispatcher::new(&settings.url, &settings.db);
+        Self {
+            service: InfluxDbService::new(dispatcher),
+        }
+    }
+
+    /// Records a new metric and dispatches it to an InfluxDB instance.
+    pub fn metric<V>(&self, measurement: Measurement, value: V, tags: Option<Tags>)
+    where
+        V: Into<influxdb::Type> + Send + 'static,
+    {
+        let mut metric = Metric::new(measurement, value.into());
+        if let Some(tags) = tags {
+            metric.with_tags(tags);
+        }
+
+        self.call(Request::from(metric))
+    }
+
+    /// Records a new event and dispatches it to an InfluxDB instance.
+    pub fn event<'a, T>(&self, title: T, description: Option<&str>, tags: Option<&'a [&'a str]>)
```

Comment:
Why do we need to specify the lifetime here?

Reply:
Ah you are right, we don't need that. It was a leftover from my previous attempts